processing/src/main/java/org/apache/druid/query/scan/ScanQueryQueryToolChest.java

```diff
@@ -131,4 +168,75 @@ public void cleanup(ScanQueryLimitRowIterator iterFromMake)
       }
     };
   }
+
+  @VisibleForTesting
+  Iterator<ScanResultValue> sortScanResultValues(Iterator<ScanResultValue> inputIterator, ScanQuery scanQuery)
+  {
+    Comparator<ScanResultValue> priorityQComparator = new ScanResultValueTimestampComparator(scanQuery);
+
+    // Converting the limit from long to int could theoretically throw an ArithmeticException but this branch
+    // only runs if limit < MAX_LIMIT_FOR_IN_MEMORY_TIME_ORDERING (which should be < Integer.MAX_VALUE)
+
+    PriorityQueue<ScanResultValue> q = new PriorityQueue<>(Math.toIntExact(scanQuery.getLimit()), priorityQComparator);
+
+    while (inputIterator.hasNext()) {
+      ScanResultValue next = inputIterator.next();
+      List<Object> events = (List<Object>) next.getEvents();
+      for (Object event : events) {
+        // Using an intermediate unbatched ScanResultValue is not that great memory-wise, but the column list
+        // needs to be preserved for queries using the compactedList result format
+        q.offer(new ScanResultValue(null, next.getColumns(), Collections.singletonList(event)));
+      }
+    }
+    // Need to convert to a List because Priority Queue's iterator doesn't guarantee that the sorted order
+    // will be maintained
+    List<ScanResultValue> sortedElements = new ArrayList<>(q.size());
+    while (q.size() != 0) {
+      sortedElements.add(q.poll());
+    }
+    return sortedElements.iterator();
+  }
+
+  /**
+   * This iterator supports iteration through any Iterable of unbatched ScanResultValues (1 event/SRV) and aggregates
+   * events into ScanResultValues with {int batchSize} events.  The columns from the first event per ScanResultValue
+   * will be used to populate the column section.
+   */
+  private static class ScanBatchedTimeOrderedIterator implements CloseableIterator<ScanResultValue>
```

Comment:
This doesn't seem to be doing anything related to time-ordering; why include "TimeOrdered" in the name?